### ISSUE TITLE
ClrConditionalModule missed import from @clr/angular 

### ISSUE THREAD

**Issue by baio** [vmware/clarity#3723]
## Describe the bug

ClrConditionalModule missed import from @clr/angular

## How to reproduce

Try to include into project 
`import { ClrConditionalModule } from '@clr/angular';`

add `import { ClrConditionalModule } from '@clr/angular';`
into project

compile project got error
`"export 'ClrConditionalModule' was not found in '@clr/angular'`

## Expected behavior

Should allow to import this module

## Versions

**App**

* Angular: [e.g. 8.0.0]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 2.1.1]

**Comment by gnomeontherun** [comment]
This is an internal module that we don't export or document, do you have a use case for why you want this?

**Comment by baio** [comment]
When importing ClrDatagridModule (not ClarityModule) expand row not working.
Please, see repro here https://stackblitz.com/edit/clr-list-expand-row-repro.

**Comment by Shijir** [comment]
This looks like a bug. We need to export the conditional modules from the datagrid module in order to enable users to use them if they are using `ClrDatagridModule`  alone without `ClarityModule `.

**Comment by Shijir** [comment]
@baio why do you need to import individual component modules? The unused modules from ClarityModule would be tree-shaken once you build your application. So you should only use ClarityModule.

**Comment by baio** [comment]
Hi @Shijir it is mostly hygienic question
1. It will bee tree shaken only in prod mode and supposedly with `build-optimizer=true` flag on (and theoretically wholesale import will takes more time to build)
2. I think it's a best practice import only what you need
3.  It is more clear from dev point of view when you see which functionality module used from another modules
4. Anyway why then clarity bother to export separated modules, just left only root one.

**Comment by gnomeontherun** [comment]
Hi @baio, Angular only does tree shaking in prod mode, that is correct. Clarity is meant to be a whole design system, not just a set of components that you consume individually. Across a large project (even if you break it into feature modules), the vast majority of Clarity is likely to be used. Angular will drop out any modules that aren't used (so if you don't use anything from Datagrid, the whole Datagrid module will be removed). 

Your point about best practice is interesting, because I believe it creates more overhead on the developer to curate the list of modules they must import when the CLI is capable of removing anything unused. The impact during development is impossible to detect to bootstrap the whole ClarityModule vs just one module. 

I'm not sure it makes it more clear to have an explicit list of modules included. Again, if the CLI can clean up what isn't used, this isn't much of a concern and frees up mental space for developers.

We do export separate modules because originally tree shaking wasn't available when we started Clarity back in early Angular 2 alpha days. At that time, importing specific modules based on your need was the only way to manage package size and that isn't true since Angular 4. The other case that is still true today is for testing, because importing the whole ClarityModule is more expensive at scale than just the specific modules needed in an isolated test (however the amount of cost is dependent upon which features you are using). Most of the time testing doesn't do a prod build before testing.

We do need to address the missing import for this module, for some reason its missing and you should only have to import ClrDatagridModule and we aren't planning to make ClrConditionalModule public at this point.

In short, this is a bug we should resolve so that you only have to import ClrDatagridModule, abd should you prefer to do this, that is something we should allow still :)

**Comment by baio** [comment]
Hey @gnomeontherun thanks for detailed response. Just to clarify a little bit more this is [where](https://github.com/scalio/ng-holistic/blob/c63e9ff47941701d07b3cedb905d4db4ef1b3870/libs/clr-list/src/lib/table/table.module.ts#L16) it just doesn't feel right (module in library which uses clarity grid)

**Comment by bdryanovski** [comment]
Is this still valid? It seems pretty old

@Shijir
